tests/SoCreate.ServiceFabric.PubSub.Tests/GivenServiceReference.cs

```diff
@@ -54,5 +54,246 @@ public void WhenDeterminingShouldDeliverMessageToActorWithUnmatchingPayload_Then
+            {
+                Customer = new
+                {
+                    Name = "Customer2"
+                }
+            }.CreateMessageWrapper();
+
+            bool shouldDeliver = actorRef.ShouldDeliverMessage(messageWrapper);
+            Assert.IsFalse(shouldDeliver);
+        }
+        [TestMethod]
+        public void WhenDeterminingShouldDeliverMessageToActorWithUnmatchingPayloadWithStartingWildCard__ThenReturnsTrue()
+        {
+            var actorRef = new ActorReferenceWrapper(new ActorReference { ActorId = ActorId.CreateRandom() }, "Customer.Name", "*1");
+            var messageWrapper = new
+            {
+                Customer = new
+                {
+                    Name = "Customer1"
+                }
+            }.CreateMessageWrapper();
+
+            bool shouldDeliver = actorRef.ShouldDeliverMessage(messageWrapper);
+            Assert.IsTrue(shouldDeliver);
+        }
+
+        /// Testing Ending Character Wild Card Support
+        [TestMethod]
+        public void WhenDeterminingShouldDeliverMessageToServiceWithMatchingPayloadWithEndingWildCard_ThenReturnsTrue()
+        {
+            var serviceRef = new ServiceReferenceWrapper(new ServiceReference(), "Customer.Name", "Customer*");
+            var messageWrapper = new
+            {
+                Customer = new
+                {
+                    Name = "Customer1"
+                }
+            }.CreateMessageWrapper();
+
+            bool shouldDeliver = serviceRef.ShouldDeliverMessage(messageWrapper);
+            Assert.IsTrue(shouldDeliver);
+        }
+
+        [TestMethod]
+        public void WhenDeterminingShouldDeliverMessageToServiceWithUnmatchingPayloadWithEndingWildCard__ThenReturnsFalse()
+        {
+            var serviceRef = new ServiceReferenceWrapper(new ServiceReference(), "Customer.Name", "Supplier*");
+            var messageWrapper = new
+            {
+                Customer = new
+                {
+                    Name = "Customer2"
+                }
+            }.CreateMessageWrapper();
+
+            bool shouldDeliver = serviceRef.ShouldDeliverMessage(messageWrapper);
+            Assert.IsFalse(shouldDeliver);
+        }
+
+        [TestMethod]
+        public void WhenDeterminingShouldDeliverMessageToActorWithUnmatchingPayloadWithEndingWildCard__ThenReturnsFalse()
+        {
+            var actorRef = new ActorReferenceWrapper(new ActorReference { ActorId = ActorId.CreateRandom() }, "Customer.Name", "Supplier*");
+            var messageWrapper = new
+            {
+                Customer = new
+                {
+                    Name = "Customer2"
+                }
+            }.CreateMessageWrapper();
+
+            bool shouldDeliver = actorRef.ShouldDeliverMessage(messageWrapper);
+            Assert.IsFalse(shouldDeliver);
+        }
+        [TestMethod]
+        public void WhenDeterminingShouldDeliverMessageToActorWithUnmatchingPayloadWithEndingWildCard__ThenReturnsTrue()
```

Comment:
Test name should be "WhenDeterminingShouldDeliverMessageToActorWithMatchingPayloadWithEndingWildCard__ThenReturnsTrue".